### ISSUE TITLE
关于__C.ARCH.SEQ_LENGTH的参数

### ISSUE THREAD

**Issue by chenhaofen** [MaybeShewill-CV/CRNN_Tensorflow#281]
作者你好。我用了你的工程去训练了自己的数据。我的数据长短不一样。最长的可能有23个字,都是手写字体。最短的可能只有2个。训练了大概16个小时左右,有些样本可以识别出来,有些不行。是训练的时间不够长还是要调一下哪里的参数呢?

**Comment by MaybeShewill-CV** [comment]
@chenhaofen 你的训练数据是英文还是中文? 图像尺寸多大?

**Comment by chenhaofen** [comment]
中文手写的。尺寸也是按照你的调好。尺寸都不一样，有大有小

**Comment by MaybeShewill-CV** [comment]
@chenhaofen 你如果用的是中文训练的config配置 图像会被统一resize到280* 32大小 sequence_length=70不需要修改. 你检查下你训练的时候哪里弄错了吧, 看看是不是config文件参数写错了:)

**Comment by chenhaofen** [comment]
# Number of units in each LSTM cell
__C.ARCH.HIDDEN_UNITS = 256
# Number of stacked LSTM cells
__C.ARCH.HIDDEN_LAYERS = 2
# Sequence length.  This has to be the width of the final feature map of the CNN, which is input size width / 4
__C.ARCH.SEQ_LENGTH = 70  # cn dataset
#__C.ARCH.SEQ_LENGTH = 25  # synth90k dataset
# Width x height into which training / testing images are resized before feeding into the network
__C.ARCH.INPUT_SIZE = (280, 32)  # cn dataset
#__C.ARCH.INPUT_SIZE = (100, 32)  # synth90k dataset
# Number of channels in images
__C.ARCH.INPUT_CHANNELS = 3
# Number character classes
__C.ARCH.NUM_CLASSES = 5984  # cn dataset 5825
#__C.ARCH.NUM_CLASSES = 37  # synth90k dataset

# Train options
__C.TRAIN = edict()

# Use early stopping?
__C.TRAIN.EARLY_STOPPING = False
# Wait at least this many epochs without improvement in the cost function
__C.TRAIN.PATIENCE_EPOCHS = 6
# Expect at least this improvement in one epoch in order to reset the early stopping counter
__C.TRAIN.PATIENCE_DELTA = 1e-3

# Set the shadownet training epochs
__C.TRAIN.EPOCHS = 2000000
# Set the display step
__C.TRAIN.DISPLAY_STEP = 1
# Set the test display step during training process
__C.TRAIN.TEST_DISPLAY_STEP = 100
# Set the momentum parameter of the optimizer
__C.TRAIN.MOMENTUM = 0.9
# Set the initial learning rate
__C.TRAIN.LEARNING_RATE = 0.01
# Set the GPU resource used during training process
__C.TRAIN.GPU_MEMORY_FRACTION = 0.9
# Set the GPU allow growth parameter during tensorflow training process
__C.TRAIN.TF_ALLOW_GROWTH = True
# Set the shadownet training batch size
__C.TRAIN.BATCH_SIZE = 16 #32
# Set the shadownet validation batch size
__C.TRAIN.VAL_BATCH_SIZE = 32
# Set the learning rate decay steps
__C.TRAIN.LR_DECAY_STEPS = 500000
# Set the learning rate decay rate
__C.TRAIN.LR_DECAY_RATE = 0.1
# Update learning rate in jumps?
__C.TRAIN.LR_STAIRCASE = True
# Set multi process nums
__C.TRAIN.CPU_MULTI_PROCESS_NUMS = 6
# Set Gpu nums
__C.TRAIN.GPU_NUM = 2
# Set moving average decay
__C.TRAIN.MOVING_AVERAGE_DECAY = 0.9999
# Set val display step
__C.TRAIN.VAL_DISPLAY_STEP = 1000

# Test options
__C.TEST = edict()

# Set the GPU resource used during testing process
__C.TEST.GPU_MEMORY_FRACTION = 0.6
# Set the GPU allow growth parameter during tensorflow testing process
__C.TEST.TF_ALLOW_GROWTH = False
# Set the test batch size
__C.TEST.BATCH_SIZE = 32

参数都是按照你的参数来调的 
__C.ARCH.NUM_CLASSES = 5984  # cn dataset 5825 字数这里和你的不一样 我的字的类别数多了一点

**Comment by MaybeShewill-CV** [comment]
@chenhaofen 那这个设置不需要修改sequence length为170：）

**Comment by chenhaofen** [comment]
这里的配置也是按照作者您的配置来的。还是说我训练的时间不太够呢?

**Comment by MaybeShewill-CV** [comment]
@chenhaofen 你不是在问sequence length参数问题么==!

**Comment by chenhaofen** [comment]
不好意思,可能我表达得不是很清楚把。我怀疑是这个的问题,听了您的解答,应该不会是的。所以我就问问是不是我的训练时间不够,导致一部分可以识别出来,一部分识别不出来

**Comment by MaybeShewill-CV** [comment]
@chenhaofen 这个就没办法很清楚的说明具体是什么原因了，很多方面都可能导致这个问题：）

**Comment by chenhaofen** [comment]
恩。好的。。我研究一下。谢谢作者

**Comment by MaybeShewill-CV** [comment]
@chenhaofen 没关系:)
